<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MyOwnQtPracticeClass</class>
 <widget class="QMainWindow" name="MyOwnQtPracticeClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>672</width>
    <height>465</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MyOwnQtPractice</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="QViewport" name="viewport" native="true">
      <property name="styleSheet">
       <string notr="true">background-color: rgb(0,0,0);</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QWidget" name="widget" native="true">
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>10</y>
         <width>142</width>
         <height>24</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout"/>
      </widget>
      <widget class="QPushButton" name="OooWhee">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>340</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>OooWhee</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>30</y>
         <width>50</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>y direction</string>
       </property>
      </widget>
      <widget class="QSlider" name="ySlider">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>30</y>
         <width>84</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-300</number>
       </property>
       <property name="maximum">
        <number>300</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>60</y>
         <width>50</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>z direction</string>
       </property>
      </widget>
      <widget class="QSlider" name="zSlider">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>60</y>
         <width>84</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-300</number>
       </property>
       <property name="maximum">
        <number>300</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QPushButton" name="ClearButton">
       <property name="geometry">
        <rect>
         <x>240</x>
         <y>360</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Clear Screen</string>
       </property>
      </widget>
      <widget class="QLabel" name="ModelLabel">
       <property name="geometry">
        <rect>
         <x>196</x>
         <y>0</y>
         <width>131</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Model Name: </string>
       </property>
      </widget>
      <widget class="QSlider" name="xSlider">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>0</y>
         <width>84</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-300</number>
       </property>
       <property name="maximum">
        <number>300</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>0</y>
         <width>50</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>x direction</string>
       </property>
      </widget>
      <widget class="QComboBox" name="ModelPicker">
       <property name="geometry">
        <rect>
         <x>198</x>
         <y>20</y>
         <width>121</width>
         <height>22</height>
        </rect>
       </property>
       <property name="currentText">
        <string>Blender</string>
       </property>
       <item>
        <property name="text">
         <string>Blender</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Cup</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Cyph</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Huble</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Planet Camera</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Shovel</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Shuttle</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Spoon</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Wrench</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Zepplin</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>80</y>
         <width>321</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>100</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Ambient R</string>
       </property>
      </widget>
      <widget class="QDial" name="AmbR">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>90</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>100</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>G</string>
       </property>
      </widget>
      <widget class="QDial" name="AmbG">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>90</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QDial" name="AmbB">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>90</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>100</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>B        </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>160</y>
         <width>51</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Specular R</string>
       </property>
      </widget>
      <widget class="QDial" name="SpecR">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>150</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>160</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>G</string>
       </property>
      </widget>
      <widget class="QDial" name="SpecG">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>150</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>160</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>B        </string>
       </property>
      </widget>
      <widget class="QDial" name="SpecB">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>150</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>130</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>G</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>130</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>B        </string>
       </property>
      </widget>
      <widget class="QDial" name="DiffB">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>120</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QDial" name="DiffR">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>120</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QDial" name="DiffG">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>120</y>
         <width>21</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>130</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Diffuse R</string>
       </property>
      </widget>
      <widget class="QPushButton" name="LightDistanceButton">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>140</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>7</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Point Light(1) = true</string>
       </property>
      </widget>
      <widget class="QComboBox" name="LightPicker">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>130</y>
         <width>69</width>
         <height>22</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Light1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Light2</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Light3</string>
        </property>
       </item>
      </widget>
      <widget class="QPushButton" name="EnableLightButton">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>110</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Enable Light(1)</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>180</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Position X</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="xBox">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>180</y>
         <width>40</width>
         <height>16</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-50</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_15">
       <property name="geometry">
        <rect>
         <x>225</x>
         <y>180</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Y</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="yBox">
       <property name="geometry">
        <rect>
         <x>233</x>
         <y>180</y>
         <width>40</width>
         <height>16</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-50</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>276</x>
         <y>180</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Z</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="zBox">
       <property name="geometry">
        <rect>
         <x>285</x>
         <y>180</y>
         <width>40</width>
         <height>16</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-50</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
      </widget>
      <zorder>layoutWidget</zorder>
      <zorder>OooWhee</zorder>
      <zorder>label_2</zorder>
      <zorder>ySlider</zorder>
      <zorder>label_3</zorder>
      <zorder>zSlider</zorder>
      <zorder>ClearButton</zorder>
      <zorder>ModelLabel</zorder>
      <zorder>xSlider</zorder>
      <zorder>label</zorder>
      <zorder>ModelPicker</zorder>
      <zorder>label_4</zorder>
      <zorder>label_5</zorder>
      <zorder>AmbR</zorder>
      <zorder>label_6</zorder>
      <zorder>AmbG</zorder>
      <zorder>label_7</zorder>
      <zorder>AmbB</zorder>
      <zorder>label_8</zorder>
      <zorder>SpecR</zorder>
      <zorder>label_9</zorder>
      <zorder>SpecG</zorder>
      <zorder>label_10</zorder>
      <zorder>SpecB</zorder>
      <zorder>label_11</zorder>
      <zorder>label_12</zorder>
      <zorder>DiffB</zorder>
      <zorder>DiffR</zorder>
      <zorder>DiffG</zorder>
      <zorder>label_13</zorder>
      <zorder>LightDistanceButton</zorder>
      <zorder>LightPicker</zorder>
      <zorder>EnableLightButton</zorder>
      <zorder>label_14</zorder>
      <zorder>label_15</zorder>
      <zorder>label_16</zorder>
      <zorder>xBox</zorder>
      <zorder>yBox</zorder>
      <zorder>zBox</zorder>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>672</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QViewport</class>
   <extends>QWidget</extends>
   <header>QViewport.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="MyOwnQtPractice.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>xSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>xChangeLevel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>157</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OooWhee</sender>
   <signal>pressed()</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>generateMeshFromFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>393</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ySlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>yChangeLevel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>zChangeLevel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ClearButton</sender>
   <signal>pressed()</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ModelPicker</sender>
   <signal>activated(QString)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>generateMeshFromFile(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>267</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AmbR</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateAmbient(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AmbG</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateAmbient(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AmbB</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateAmbient(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SpecR</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateSpecular(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SpecG</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateSpecular(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SpecB</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateSpecular(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DiffR</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateDiffuse(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DiffG</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateDiffuse(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DiffB</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateDiffuse(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LightDistanceButton</sender>
   <signal>pressed()</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>toggleLightDistance()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>172</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>EnableLightButton</sender>
   <signal>pressed()</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>toggleEnableLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LightPicker</sender>
   <signal>activated(QString)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>selectLight(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateLightCoord(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>214</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateLightCoord(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MyOwnQtPracticeClass</receiver>
   <slot>updateLightCoord(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>314</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>xChangeLevel(int)</slot>
  <slot>generateMeshFromFile()</slot>
  <slot>yChangeLevel(int)</slot>
  <slot>zChangeLevel(int)</slot>
  <slot>clear()</slot>
  <slot>generateMeshFromFile(QString)</slot>
  <slot>updateAmbient(int)</slot>
  <slot>updateSpecular(int)</slot>
  <slot>updateDiffuse(int)</slot>
  <slot>toggleLightDistance()</slot>
  <slot>toggleEnableLight()</slot>
  <slot>selectLight(QString)</slot>
  <slot>updateLightCoord(int)</slot>
 </slots>
</ui>
